components/mission_key_management/client.c

```diff
@@ -4,20 +4,59 @@
 
 #include <stdlib.h>
 #include <string.h>
+#include <stdint.h>
 
-#include "client.h"
+#include <sys/socket.h>
+#include <unistd.h>
+#include <stdio.h>
 
-#ifndef CN_ENV
+//SYSTEM_HEADERS
+// ^^^ magic string used during preprocessing - do not move / remove 
+
+# include "client.h" 
+
+#if ! defined(CN_ENV)
+# include <sys/types.h>
 # include <sys/epoll.h>
-# include <sys/socket.h>
-# include <unistd.h>
-# include <stdio.h>
-#else
+#else  
 # include "cn_stubs.h"
+# include "cn_array_utils.h"
 #endif
 
-
-uint32_t client_state_epoll_events(enum client_state state) {
+// TODO `Alloc` construct not supported by `cn test`
+#if defined(CN_ENV) && ! defined(CN_TEST) 
+# include "cn_memory.h"
+#endif 
+
+#if defined(CN_TEST) 
+# define NULL ((void *)0)
+#endif 
+
+/* TODO list: 
```

Comment:
I am assuming you are still actively working on this TODO?
Also, this list should be a part of the PR description rather than being hidden in the code.

Reply:
These are done, removed